test/extended/router/externalCertificate.go

```diff
@@ -0,0 +1,682 @@
+						checkRouteStatus(oc, routes[i].Name, corev1.ConditionFalse, "ExternalCertificateValidationFailed")
+					}
+				})
+			})
+
+			g.Context("and the route is updated", func() {
+				var (
+					routeToTest   *routev1.Route
+					newSecretName = "new-ext-crt"
+				)
+
+				g.BeforeEach(func() {
+					// These tests donot *explicity* need verification on multiple routes.
+					// Hence taking only one route into account.
+					routeToTest = routes[0]
+				})
+
+				g.Context("to use new external certificate", func() {
+					g.BeforeEach(func() {
+						g.By("Creating a new secret")
+						secret, rootDerBytes, err = createSecret(oc.Namespace(), newSecretName, corev1.SecretTypeTLS, hosts...)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						_, err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Create(context.Background(), secret, metav1.CreateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						g.By("Updating the exisiting role to add RBAC permissions for the new secret")
+						newRule := fmt.Sprintf(`{"apiGroups": [""],"resources": ["secrets"],"resourceNames":["%s"],"verbs": ["get", "list", "watch"]}`, newSecretName)
+						rolePatch := fmt.Sprintf(`{"rules": [%s]}`, newRule)
+						_, err = oc.KubeClient().RbacV1().Roles(oc.Namespace()).Patch(
+							context.Background(), SECRET_READER_ROLE, types.MergePatchType, []byte(rolePatch), metav1.PatchOptions{},
+						)
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						g.By("Updating the route to use new external certificate")
+						routePatch := fmt.Sprintf(`{"spec":{"tls":{"externalCertificate":{"name":"%s"}}}}`, newSecretName)
+						_, err = oc.RouteClient().RouteV1().Routes(oc.Namespace()).Patch(
+							context.Background(), routeToTest.Name, types.MergePatchType, []byte(routePatch), metav1.PatchOptions{},
+						)
+						o.Expect(err).NotTo(o.HaveOccurred())
+					})
+
+					g.It("then also the route is reachable", func() {
+						g.By("Sending https request")
+						hostName, err := getHostnameForRoute(oc, routeToTest.Name)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						resp, err := httpsGetCall(hostName, rootDerBytes)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						o.Expect(resp).Should(o.ContainSubstring("Hello OpenShift"))
+					})
+				})
+
+				g.Context("to use new external certificate, but RBAC permissions are not added", func() {
+					g.BeforeEach(func() {
+						g.By("Creating a new secret")
+						secret, _, err = createSecret(oc.Namespace(), newSecretName, corev1.SecretTypeTLS, hosts...)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						_, err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Create(context.Background(), secret, metav1.CreateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+					})
+					g.It("route update is rejcted", func() {
+						g.By("Updating the route to use new external certificate")
+						routePatch := fmt.Sprintf(`{"spec":{"tls":{"externalCertificate":{"name":"%s"}}}}`, newSecretName)
```

Comment:
similar.

Reply:
Moved to `patchRouteWithExternalCertificate`